superset/db_engine_specs/presto.py

```diff
@@ -817,13 +818,17 @@ def _partition_query(  # pylint: disable=too-many-arguments,too-many-locals
 
         presto_version = database.get_extra().get("version")
 
+        partdollarqry = f"SHOW PARTITIONS FROM {table_name}"
+        if not presto_version or StrictVersion(presto_version) >= StrictVersion("0.199"):
+            if schema and len(schema) > 0:
```

Comment:
This logic can then be simplified as `str(table)` and thus no if/else statement is requirement.